coredns/main.go

```diff
@@ -105,13 +107,23 @@ var directives = []string{
 	"on",
 }
 
-var version = "not-compiled-properly"
+var (
+	version     = "not-compiled-properly"
+	submVersion = false
+)
 
 func init() {
+	flag.BoolVar(&submVersion, "subm-version", submVersion, "Show version")
 	lighthouse.Version = version
 	dnsserver.Directives = directives
 }
 
 func main() {
+	flag.Parse()
+	if submVersion {
+		fmt.Printf("submariner-lighthouse-coredns version: %s", version)
+		return
+	}
```

Comment:
We also log the version [here](https://github.com/submariner-io/lighthouse/blob/6cab076201b9d77adeb8714d34d54d66324d8e56/coredns/plugin/setup.go#L74). How about we just print the version here and exit if the version was requested? Same with the agent.

```suggestion
        fmt.Printf("submariner-lighthouse-coredns version: %s", version)
        
	if showVersion {
		return
	}
```

Reply:
Yeah, I thought about printing and not using loger, but what if someone is using an external logger? printf will not show up in that, right?

Reply:
How would someone use/setup an external logger? But regardless, wouldn't `fmt.Printf` go to the pod log no matter what? We'll scrape from that.

Reply:
Tried it out, `fmt.Printf` didn't show up in logs, only if we use logger. So will continue with current approach. Changed `submVersion` to `showVersion`.

Reply:
That’s odd. So it shows up only when you run it with --version? What’s the diff? Aren’t you scraping the logs in the same manner?

Reply:
I looked at **klog** code and it writes to stderr by default, ie the `logtostderr` flag defaults to true. In fact there’s no option to write to stdout. **admiral**'s zerolog also writes to [stderr](https://github.com/submariner-io/admiral/blob/6febc35a01d2fe109fa46d8432edc55dc5a33332/pkg/log/kzerolog/kzerolog.go#L78). I think that's b/c `kubectl` scrapes the log from stderr. So if we do `fmt.Fprintf(os.Stderr, …)`, it should show up in the pod logs.

Reply:
@tpantelis Okay, print to stderr worked. But two issues with it:

1. Doesn't work for `coredns` For some reason any output to stdout or stderr before plugin directive is loaded results in CrashLoopBackoff on coredns. So we can't print before plugin is loaded and if we do that `--version` will not work coz coredns is already running so it will fail to bring it up again.
2. Using print to stderr shows up in logs but without timestamp or line of code information:

```
submariner-lighthouse-agent version: devel-e7cead11fe84
2023-08-03T05:42:07.414Z INF ..e/pkg/agent/main.go:105 main                 Arguments: [/usr/local/bin/lighthouse-a
gent -alsologtostderr -v=2]
```

Am okay either ways but do prefer to use logger for log file so that version information shows up with timestamp et al in the logs. Thoughts?

Reply:
> Doesn't work for coredns For some reason any output to stdout or stderr before plugin directive is loaded results in CrashLoopBackoff on coredns. 

That's weird. Do you get a stack trace? We can log it where we do now.

> Am okay either ways but do prefer to use logger for log file so that version information shows up with timestamp et al in the logs. Thoughts?

I think it's fine w/o timestamp. It's not really needed since it's the first log entry and the pertinent info is the version.  But if you want it then set up the logger beforehand.

Reply:
> That's weird. Do you get a stack trace? We can log it where we do now.

No stack trace. Only error about missing `dns` directive.